src-frontend/app/controllers/procedure.js

```diff
@@ -44,17 +57,20 @@ export default Ember.Controller.extend({
         updateSortOrder: function(pageModels) {
             var controller = this;
 
+            pageModels = pageModels || controller.generateNewPageIndices()
```

Comment:
I know this is a JS idiom, but it is incredibly confusing and is missing a semicolon.